test/langtools/tools/javac/processing/model/element/TestListPackageFromAPI.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2017, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8278930
+ * @summary Check that when a package has Elements listed from both from classes and sources,
+ *          and then a nested class is completed, it is not first completed from source via
+ *          its enclosing class, and then again for itself.
+ * @library /tools/javac/lib
+ * @modules java.compiler
+ *          jdk.compiler
+ * @run main TestListPackageFromAPI
+ */
+
+import com.sun.source.util.JavacTask;
+import java.io.IOException;
+import java.io.Writer;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import javax.lang.model.element.PackageElement;
+import javax.tools.DiagnosticListener;
+import javax.tools.ForwardingJavaFileManager;
+import javax.tools.JavaFileManager;
+import javax.tools.JavaFileObject;
+import javax.tools.JavaFileObject.Kind;
+import javax.tools.SimpleJavaFileObject;
+import javax.tools.StandardLocation;
+import javax.tools.ToolProvider;
+
+public class TestListPackageFromAPI {
+
+    public static void main(String... args) throws IOException, URISyntaxException, InterruptedException {
+        Path base = Paths.get(".");
+        Path src = base.resolve("src");
+        Path classes = base.resolve("classes");
+
+        Files.createDirectories(src);
+        Files.createDirectories(classes);
+
+        Files.newOutputStream(classes.resolve("Test$Nested1.class")).close();
+        Files.newOutputStream(classes.resolve("Test.class")).close();
+
+        Thread.sleep(2000);
```

Comment:
Mild ugh

Reply:
Thanks for the comment!

It was waiting to reliably make the timestamps different, but I've cleaned up the test here to avoid that:
https://github.com/openjdk/jdk18/pull/85/commits/6bef38f54e791092eb88843c766d8f5d064ac9c5